File: api/schemas/actors.py
# api/schemas/actors.py
from __future__ import annotations
from pydantic import BaseModel, Field
from api.schemas.status import DataMode


class ActorItem(BaseModel):
    id: str
    name: str
    party: str = ""
    party_color: str = "#94A3B8"
    role: str = ""
    bio: str = ""
    exposure: int = 0     # 0-100
    approval: int = 50    # 0-100
    sentiment: str = "stable"  # up | down | stable


class ActorsResponse(BaseModel):
    actors: list[ActorItem] = Field(default_factory=list)
    total: int
    mode: DataMode

File: apps/web/lib/types/actors.ts
// Actor and Party types — mirrors the shape from apps/web/app/actores/page.tsx
export interface Actor {
  id: string;
  name: string;
  party: string;
  partyColor: string;
  role: string;
  bio: string;
  exposure: number;
  approval: number;
  sentiment: "up" | "down" | "stable";
}

export interface Party {
  code: string;
  color: string;
}
